sdk/storage/azure-storage-common/inc/azure/storage/common/storage_exception.hpp

```diff
@@ -38,5 +38,13 @@ namespace Azure { namespace Storage {
      */
     static StorageException CreateFromResponse(
         std::unique_ptr<Azure::Core::Http::RawResponse> response);
+
+  private:
+    StorageException(
+        const std::string& whatArg,
+        std::unique_ptr<Azure::Core::Http::RawResponse> rawResponse)
+        : RequestFailedException(whatArg, std::move(rawResponse))
```

Comment:
We were planning on removing this 2-param ctor from RFE. Is calling it necessary for this case?

https://github.com/Azure/azure-sdk-for-cpp/issues/2543